pallets/asset/src/ethereum.rs

```diff
@@ -4,33 +4,10 @@
 // https://github.com/paritytech/polkadot/blob/013c4a8041e6f1739cc5b785a2874061919c5db9/runtime/common/src/claims.rs#L248-L251
 
 use codec::{Decode, Encode};
+use pallet_asset_types::{EcdsaSignature, EthereumAddress};
 use sp_io::{crypto::secp256k1_ecdsa_recover, hashing::keccak_256};
-#[cfg(feature = "std")]
-use sp_runtime::{Deserialize, Serialize};
 use sp_std::vec::Vec;
 
-/// An Ethereum address (i.e. 20 bytes, used to represent an Ethereum account).
-///
-/// This gets serialized to the 0x-prefixed hex representation.
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-#[derive(Clone, Copy, PartialEq, Eq, Encode, Decode, Default, Debug)]
-pub struct EthereumAddress(pub [u8; 20]);
```

Comment:
Not sure why things in this file were moved out. If not necessary to break a cyclic dependency (where, if so?), I'd prefer that we keep the Ethereum bits accumulated here.

Reply:
It gets moved because of cyclic dependencies but I think we can manage to keep these types here without creating the cyclic dependencies. btw this is my doing when I tried to solve the cyclic dependencies anyway @ParnianAlimi could you please take a look and let me know if you have face any difficulties.

Reply:
Right, I didn't see any references to these types outside of the assets pallet, so it should be possible to keep them where they were.

Reply:
No worries, should be straight forward to bring them back.

Reply:
As a side note, `EthereumAddress` is duplicated in `pallets/asset/src/ethereum.rs` and `pallets/asset/src/lib.rs`, I'll just bring it back to what it was before and leave it to you to remove one.

Reply:
Oops; can you please remove the one in `lib.rs`?